src/adapter/src/explain_new/mir/text.rs

```diff
@@ -294,6 +300,12 @@ impl<'a> Displayable<'a, MirRelationExpr> {
                     Ok(())
                 })?;
             }
+            Join {
+                implementation: JoinImplementation::IndexedFilter(id, _key, literal_constraints),
```

Comment:
I'm not sure that this will work nicely in `WITH(linear_chains)` mode. See how the rendering of other single-input nodes (like `Map` or `Filter`) is conditioned on this property in this file.